pywhy_graphs/algorithms/generic.py

```diff
@@ -333,3 +334,207 @@ def _single_shortest_path_early_stop(G, firstlevel, paths, cutoff, join, valid_p
                     nextlevel[w] = 1
         level += 1
     return paths
+
+
+def _find_directed_parents(G, node):
+    """Finds the parents of a node in the directed and the bidirected subgraphs.
+
+    Parameters
+    ----------
+    G : Graph
+        The graph.
+    node : node label
+        The node for which we have to find the parents.
+
+    Returns
+    -------
+    out : set
+        The parents of the provided node.
+    """
+
+    if not isinstance(G, CPDAG):
+        bidirected_parents = set(G.sub_bidirected_graph().neighbors(node))
+    directed_parents = set(G.sub_directed_graph().predecessors(node))
+
+    out = bidirected_parents.union(directed_parents)
+
+    return out
+
+
+def _find_directed_children(G, node):
+    """Finds the children of a node in the directed and the bidirected subgraphs.
+
+    Parameters
+    ----------
+    G : Graph
+        The graph.
+    node : node label
+        The node for which we have to find the children.
+
+    Returns
+    -------
+    out : set
+        The children of the provided node.
+    """
+    if not isinstance(G, CPDAG):
+        bidirected_children = set(G.sub_bidirected_graph().neighbors(node))
+    directed_children = set(G.sub_directed_graph().successors(node))
+
+    out = bidirected_children.union(directed_children)
+
+    return out
+
+
+def _is_collider(G, node):
+    """Checks if the given node is a collider or not.
+
+    Parameters
+    ----------
+    G : graph
+        The graph.
+    node : node
+        The node to be checked.
+
+    Returns
+    -------
+    iscollider : bool
+        Bool is set true if the node is a collider, false otherwise.
+    """
+    parents = _find_directed_parents(G, node)
+
+    if len(parents) > 1:
+        return True
+    return False
```

Comment:
The way a collider is defined is given (X, Y, Z), Y is a collider in G if X and Y are adjacent, and Y and Z are adjacent and X *-> Y and Y <-* Z.

Reply:
I am not sure what is wrong with this implementation. Is it the case that it can only be called a collider if and only if there are 2 nodes that have a directed edge pointing towards the third node?

Reply:
Yep! A collider is a collider if there is two arrow endpoints. Usually it is defined as a function of three nodes. 

So 

Is_collider(G, a,b,c)

Checks if say b is a collider by determining if both a and c have an arrow pointing to b. This can be via a directed edge or bidirected edge.